internal/controllers/machinedeployment/machinedeployment_sync_test.go

```diff
@@ -816,6 +823,11 @@ func assertMachineSet(g *WithT, actualMS *clusterv1.MachineSet, expectedMS *clus
 
 	// Check MachineTemplateSpec
 	g.Expect(actualMS.Spec.Template.Spec).Should(BeComparableTo(expectedMS.Spec.Template.Spec))
+
+	// Check MachineNamingStrategy
+	if actualMS.Spec.MachineNamingStrategy != nil {
+		g.Expect(actualMS.Spec.MachineNamingStrategy.Template).Should(BeComparableTo(expectedMS.Spec.MachineNamingStrategy.Template))
+	}
```

Comment:
```suggestion
	g.Expect(actualMS.Spec.MachineNamingStrategy.Template).Should(BeComparableTo(expectedMS.Spec.MachineNamingStrategy.Template))
```
Can we simply do this instead? If not, why not?